frontend/src/components/authentication/CreateAccountForm.jsx

```diff
@@ -0,0 +1,331 @@
+import React, { Component } from "react";
+import LOCALIZE from "../../text_resources";
+import validateName, { validateEmail, validatePassword } from "../../helpers/regexValidator";
+import "../../css/create-account-form.css";
+
+const styles = {
+  createAccountContent: {
+    padding: "12px 32px 0 32px"
+  },
+  inputTitle: {
+    padding: "12px 0 6px 0",
+    fontWeight: "bold"
+  },
+  inputs: {
+    width: "100%",
+    padding: "3px 6px 3px 6px",
+    borderRadius: 4
+  },
+  inputForNames: {
+    width: 190,
+    padding: "3px 6px 3px 6px",
+    borderRadius: 4
+  },
+  iconForNames: {
+    color: "#278400",
+    position: "absolute",
+    margin: "8px 0 0 170px"
+  },
+  iconForOtherFields: {
+    color: "#278400",
+    position: "absolute",
+    margin: "8px 0 0 370px"
+  },
+  loginBtn: {
+    width: 150,
+    display: "block",
+    margin: "24px auto"
+  },
+  validationError: {
+    color: "red",
+    marginTop: 6
+  }
+};
+
+let SUBMIT_BTN_DISABLED;
+
+class CreateAccountForm extends Component {
+  state = {
+    isFirstLoad: true,
+    isValidFirstName: false,
+    isValidLastName: false,
+    isValidEmail: false,
+    isValidPassword: false,
+    isFirstPasswordLoad: true,
+    isValidPasswordConfirmation: false
+  };
+
+  firstNameValidation = () => {
+    const firstName = document.getElementById("first-name").value;
+    const isValide = validateName(firstName);
+    this.setState({ isFirstLoad: false });
+    if (isValide) {
+      this.setState({ isValidFirstName: true });
+    } else {
+      this.setState({ isValidFirstName: false });
+    }
+  };
+
+  lastNameValidation = () => {
+    const lastName = document.getElementById("last-name").value;
+    const isValide = validateName(lastName);
+    this.setState({ isFirstLoad: false });
+    if (isValide) {
+      this.setState({ isValidLastName: true });
+    } else {
+      this.setState({ isValidLastName: false });
+    }
+  };
+
+  emailValidation = () => {
+    const email = document.getElementById("email").value;
```

Comment:
These use all grab the element by ID to get the value instead of storing the value in state. We looked at this a bit on Wednesday, but here's the link to how we'd store the value in state: https://reactjs.org/docs/forms.html

This might be a bit of a re-factor, so let me know if you want to pair on it.

Reply:
Fixed in commit #https://github.com/code-for-canada/project-thundercat/pull/156/commits/4c28908545c05081a121d3707a389e4821a3bb33